### DISCUSSION TITLE
TLS secret specification

### DISCUSSION THREAD

**Discussion by jakubdyszkiewicz** [kubernetes-sigs/gateway-api#1059] [General]
Hey,

In the Listener TLS spec, there is a field called [`CertificateRefs`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.SecretObjectReference). However, it is not specified what is the standard format that should be supported.

```
CertificateRefs can reference to standard Kubernetes resources, i.e. Secret, or implementation-specific custom resources.

Support: Core - A single reference to a Kubernetes Secret
Support: Implementation-specific (More than one reference or other resource types)
```

Should we support Secrets of a type `kubernetes.io/tls` as Core support?

Thanks!

**Comment by youngnick** [comment]
Thanks for this @jakubdyszkiewicz, we should definitely indicate what types are required.

**Comment by hbagdi** [comment]
Yes, the intention of `Support: Core - A single reference to a Kubernetes Secret` was to support `kubernetes.io/tls` since that is a standard that other k8s projects understand (such as cert-manager). We need to use the standard to make sure that the feature is portable (requirement for the core conformance level).

@jakubdyszkiewicz Would you be up for sending a PR to clarify that?

**Comment by jakubdyszkiewicz** [comment]
Created a PR https://github.com/kubernetes-sigs/gateway-api/pull/1084
